File: Recipe - Java/app/src/main/java/com/example/recipe_java/ui/foodstuff/FoodstuffAdapter.java
package com.example.recipe_java.ui.foodstuff;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.recipe_java.R;
import com.example.recipe_java.models.Foodstuff;
import org.jetbrains.annotations.NotNull;
import java.util.ArrayList;

public class FoodstuffAdapter extends RecyclerView.Adapter<FoodstuffAdapter.FoodstuffViewHolder> {

    private ArrayList<Foodstuff> foodstuffs;

    public void setFoodstuff(ArrayList<Foodstuff> foodstuffs) {
        this.foodstuffs = foodstuffs;

        notifyDataSetChanged();
    }

    @NonNull
    @NotNull
    @Override
    public FoodstuffViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_foodstuff, parent, false);
        return new FoodstuffViewHolder(view);
    }

    @Override
    public int getItemCount() {
        return foodstuffs.size();
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void onBindViewHolder(@NonNull @NotNull FoodstuffViewHolder holder, int position) {
        Context context = holder.itemView.getContext();
        Foodstuff foodstuff = foodstuffs.get(position);
        holder.foodstuffName.setText(foodstuff.getName());
        holder.nutriValues.setText(context.getString(R.string.label_nutri_values) +
                context.getString(R.string.label_calories) + Foodstuff.convertKcalToCal(foodstuff.getCalories()) + "kcal, " +
                context.getString(R.string.label_protein) + foodstuff.getProtein()  + "g, " +
                context.getString(R.string.label_fats) + foodstuff.getFats() + "g, ");
    }

    class FoodstuffViewHolder extends RecyclerView.ViewHolder {

        TextView foodstuffName;
        TextView nutriValues;

        public FoodstuffViewHolder(@NonNull @NotNull View itemView) {
            super(itemView);
            foodstuffName = itemView.findViewById(R.id.foodstuffNameText);
            nutriValues = itemView.findViewById(R.id.nutriValuesText);
        }
    }
}

File: Recipe - Kotlin/app/src/main/java/com/example/recipe/ui/foodstuff/FoodstuffAdapter.kt
package com.example.recipe.ui.foodstuff

import android.annotation.SuppressLint
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.recipe.R
import com.example.recipe.models.Foodstuff

class FoodstuffAdapter: RecyclerView.Adapter<FoodstuffAdapter.FoodstuffViewHolder>() {

    private var foodstuffs: MutableList<Foodstuff> = mutableListOf()

    fun setFoodstuff(foodstuff: MutableList<Foodstuff>) {
        this.foodstuffs = foodstuff
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FoodstuffViewHolder {
        val view: View = LayoutInflater.from(parent.context).inflate(R.layout.item_foodstuff, parent, false)
        return FoodstuffViewHolder(view)
    }

    @SuppressLint("SetTextI18n")
    override fun onBindViewHolder(holder: FoodstuffViewHolder, position: Int) {
        val context = holder.itemView.context
        val foodstuff = foodstuffs[position]
        holder.foodstuffName.text = foodstuff.name
        holder.nutriValues.text = "${context.getString(R.string.label_nutri_values)} " +
                "${context.getString(R.string.label_calories)} ${Foodstuff.convertKcalToCal(foodstuff.calories)}kcal, " +
                "${context.getString(R.string.label_protein)} ${foodstuff.protein}g, " +
                "${context.getString(R.string.label_fats)} ${foodstuff.fats}g"
    }

    override fun getItemCount(): Int {
        return foodstuffs.size
    }

    inner class FoodstuffViewHolder(itemView: View): RecyclerView.ViewHolder(itemView) {
        var foodstuffName: TextView = itemView.findViewById(R.id.foodstuffNameText)
        var nutriValues: TextView = itemView.findViewById(R.id.nutriValuesText)
    }
}
